### ISSUE TITLE
RSSI calibration code is incorrect

### ISSUE THREAD

**Issue by wutje** [DualTachyon/uv-k5-firmware#62]
I have been looking at the RSSI calib code and it the current implementation does not make sense, at all.
The current implementation is plain wrong. The UI code uses Band to index the gEEPROM_RSSI_CALIB array:
```
        if (RSSI >= gEEPROM_RSSI_CALIB[gRxVfo->Band][3]) {
		Level = 6;
	} else if (RSSI >= gEEPROM_RSSI_CALIB[gRxVfo->Band][2]) {
		Level = 4;
	} else if (RSSI >= gEEPROM_RSSI_CALIB[gRxVfo->Band][1]) {
		Level = 2;
	} else if (RSSI >= gEEPROM_RSSI_CALIB[gRxVfo->Band][0]) {
		Level = 1;
	} else {
		Level = 0;
	}
```

But the array is only 3 large
```
uint16_t gEEPROM_RSSI_CALIB[3][4];
```

Meaning for VHF and up we read whatever the compiler&linker put up there in RAM. A classic out-of-bound access .

However looking through the EEPROM code that inits the gEEPROM_RSSI_CALIB array I found this:
```	
        EEPROM_ReadBuffer(0x1EC0, gEEPROM_1EC0_0, 8);
	memcpy(gEEPROM_1EC0_1, gEEPROM_1EC0_0, 8);
	memcpy(gEEPROM_1EC0_2, gEEPROM_1EC0_0, 8);
	memcpy(gEEPROM_1EC0_3, gEEPROM_1EC0_0, 8);

	EEPROM_ReadBuffer(0x1EC8, gEEPROM_RSSI_CALIB[0], 8);
	memcpy(gEEPROM_RSSI_CALIB[1], gEEPROM_RSSI_CALIB[0], 8);
	memcpy(gEEPROM_RSSI_CALIB[2], gEEPROM_RSSI_CALIB[0], 8);
```
Knowing how the gEEPROM_RSSI_CALIB is used. Data is copied to 7 blocks of 8 bytes / 4 uint16_t.
I think the gEEPROM_RSSI_CALIB is actually 7 long. 
With the first 4 being equal.
And the last 3 being equal.

But somehow during reverse engineering this got mixed up?

**Comment by OneOfEleven** [comment]
> @OneOfEleven, If you ever wondered about dodgy RSSI.

Yes !  it's not made that much sense to me either. There are actually 8 16-bit values in all, but we're only the upper 4 16-bit values. The 8 values are stored at eeprom 0x1EC0

**Comment by OneOfEleven** [comment]
My values are ..

1EC0   6E 00   78 00   82 00   8C 00   B4 00   BE 00   C8 00   D2 00

**Comment by OneOfEleven** [comment]
Maybe it's the 2 bands (VHF and UHF) that the radio cover.

**Comment by wutje** [comment]
> Actually I misread my own screen shot. I know what's going on!
> It is a bug in my code.
> 
> The 1EC0 calibration is right after gRSSI_CALIB. So really what the code was doing and was not obvious to me is:
> 
> read(&gCalib[0], 0x1EC8); + memcpy to repeat slots
> read(&gCalib[3], 0x1EC0); + memcpy to repeat slots
> 
> There is just one structure rather than 2. @wutje had the right instinct, but it wasn't obvious to me without looking at the memory layout.
> 
> I'll fix it in a moment.
> 
> 
> 

This makes the UI code work properly. Sorry if I was not clear enough at the beginning.

Did you also update the eeprom content  excel sheet? Nice piece of work btw.

**Comment by OneOfEleven** [comment]
Well done both

**Comment by OneOfEleven** [comment]
Thank you very much !
